src/main/java/mrjake/aunis/tileentity/stargate/StargateClassicBaseTile.java

```diff
@@ -180,12 +181,23 @@ public void onLoad() {
 			updatePowerTier();
 		}
 	}
+
+	private BlockPos lastPos = BlockPos.ORIGIN;
 	
 	@Override
 	public void update() {
 		super.update();
 		
 		if (!world.isRemote) {
+			if (!lastPos.equals(pos)) {
+				lastPos = pos;
+				generateAddresses(!hasUpgrade(StargateClassicBaseTile.StargateUpgradeEnum.CHEVRON_UPGRADE));
```

Comment:
What would you recommend to check whether the tile has moved?

Reply:
Btw, does it actually matter? The address will stay the same, since the generator is using the position as seed.

Reply:
Slava mentions custom Stargate addresses, so someone using commands won't be happy with those changes.

Reply:
Maybe save a boolean to NBT like `addressOverriddenByCommand` or just `fixedAddress` with a docstring indicating it handles address persistence when used with commands?

Reply:
Ah, I missed that command. I've added lastPos assignment in `onLoad`, and I think it works now.

Reply:
Did you check that? With command and with Warp drive or other gate-moving mods?

Reply:
Yes

Reply:
Not sure about that. I think now code inside `!lastPos.equals(pos)` condition will not be called at all. Because position will be set in `onLoad` method after TileEntity creation and it's not possible that this position will not be equal to the current gate position

Reply:
That's what I'm thinking. `onLoad` executes before `update` and whole effort of making this mod WarpDrive compatible is worthless.